File: string/decode-the-message/decode-the-message.go
package decodethemessage

func decodeMessage(key string, message string) string {
	dic := map[rune]rune{
		' ': ' ',
	}

	letter := 'a'
	for _, c := range key {
		_, ok := dic[c]
		if c == ' ' || ok {
			continue
		}

		dic[c] = letter
		letter++
	}

	ans := make([]rune, len(message))

	for i, c := range message {
		ans[i] = dic[c]
	}

	return string(ans)
}

File: string/decode-the-message/decode-the-message.ts
const A_CODE = 'a'.charCodeAt(0);

function decodeMessage(key: string, message: string): string {
    const dic = {' ': ' '};
    let code = A_CODE;
    for (let c of key) {
        if (c === ' ' || dic[c]) continue;
        const newKey = String.fromCharCode(code++);
        dic[c] = newKey;
    }

    const ans = Array<string>(message.length)
    for (let i = 0; i < message.length; i++) {
        ans[i] = dic[message[i]];
    }

    return ans.join('');
};
